### ISSUE TITLE
bucket4j+hazelcast ClassNotFound exception

### ISSUE THREAD

**Issue by Puneetsri** [vladimir-bukhtoyarov/bucket4j#57]
Hello,
I'm using JHipster(4.14.0) which uses both hazelcast[3.9.1]+bucket4j[3.0.2] for implementing Distributed cache and rate-limiting. But I am blocked by 'ClassNotFound' exception being thrown from Hazelcast for a bucket4j class [io.github.bucket4j.grid.jcache.ExecuteProcessor].

In my project I have the bucket4j-core & bucket4j-jcache present which does contain the class hazelcast is complaining about [io.github.bucket4j.grid.jcache.ExecuteProcessor],

I've created an issue towards JHipster https://github.com/jhipster/generator-jhipster/issues/7178  but probably issue is between hazelcast and bucket4j.
Not sure if this is related to https://github.com/hazelcast/hazelcast/issues/8972.

**Comment by vladimir-bukhtoyarov** [comment]
@Puneetsri check that all hazelcast nodes contain both bucket4j-core and bucket4j-jcache jars in its classpath.

**Comment by vladimir-bukhtoyarov** [comment]
If JHipster starts Hazelcast on dedicated JVMs then bucket4j jars must be added to classpath of both Gateway and Hazelcast nodes.

**Comment by Puneetsri** [comment]
@vladimir-bukhtoyarov No both Gateway and Hazelcast are on the same JVM.

**Comment by vladimir-bukhtoyarov** [comment]
Looks like Hazelcast problem https://github.com/hazelcast/hazelcast/issues/8972#issuecomment-368982599

**Comment by Puneetsri** [comment]
This is what i as well mentioned in the problem description, might be related. Should i create a separate issue for them?

**Comment by vladimir-bukhtoyarov** [comment]
I suppose just add your use case to the hazelcast/hazelcast#8972

**Comment by Puneetsri** [comment]
Thanks, Both this issue and the original is mentioned in hazelcast/hazelcast#8972

**Comment by vladimir-bukhtoyarov** [comment]
@Puneetsri if you will provide the scenario to reproduce(with detalization level enough for user which never worked with jhipster) then I can take a look to the problem, to be sure that problem in the hazelcast#8972

**Comment by Puneetsri** [comment]
Hello @vladimir-bukhtoyarov , Original Issue  https://github.com/jhipster/generator-jhipster/issues/7178 has just been fixed by Julien in JHipster, I have yet to test it once it is available. So I'm closing this issue. Thanks.

**Comment by vladimir-bukhtoyarov** [comment]
nice
